cdap-proto/src/main/java/co/cask/cdap/proto/NamespaceConfig.java

```diff
@@ -28,14 +29,42 @@
   @SerializedName("scheduler.queue.name")
   private final String schedulerQueueName;
 
-  public NamespaceConfig(String schedulerQueueName) {
+  @SerializedName("hdfs.directory")
+  private final String hdfsDirectory;
+
+  @SerializedName("hbase.namespace")
+  private final String hbaseNamespace;
+
+  @SerializedName("hive.database")
```

Comment:
Yea - I guess it just depends what we want the user to see in the json that they pass in:
`hive.database`
or
`hiveDatabase`
I'm fine with either.